public/js/src/module/records-queue.js

```diff
@@ -60,26 +64,35 @@ function set( record ) {
             }
 
             return store.record.set( record );
-        } )
-        .then( _updateRecordList );
+        } );
 }
 
 /**
- * Updates an existing record
+ * Creates (sets) or updates a record.
+ *
+ * @param { 'set' | 'update' } action - determines whether to create or update the record
+ * @param { Record } record - the record to save
  *
- * @param  { object } record - a record object
- * @return { Promise } a promise that resolves with undefined
+ * @return { Promise<undefined> }
  */
-function update( record ) {
-    return store.record.update( record )
-        .then( _updateRecordList );
+function save( action, record ) {
```

Comment:
Ahhh, thanks for clarifying, I didn’t follow what dynamic call you were referring to at the time. I strongly agree with making that more explicit.

I think we’ll need @MartijnR to at some point help us detangle set and update. It’s not clear to me they even need to be separate.

Reply:
This is something which won’t be such an issue if we move toward TypeScript, because you can actually use language services to track even dynamic calls like that. (That changes/refactor doc is still in progress! I’m gonna stay away from code this weekend but if I feel like filling some idle time that would be a good way to do it.)

But otherwise there’s basically no tooling besides exhaustive regex/pattern based search that can find calls like that. I’m just grateful these projects use modules properly. It wasn’t long ago I worked in a MeteorJS project where even imports were implicit, good luck finding **anything** with confidence 🤯